src/components/core/posts/SinglePost.tsx

```diff
@@ -22,6 +22,35 @@ export default function SinglePost({
     await navigator.clipboard.writeText(color);
   };
 
+  function createdAtDateFormatter(postCreationTime:string){
```

Comment:
Looking good but can we make an object with all the checks and iterate over it. Currently, as you can see lots of variables are being made so it might increase the space complexity

Reply:
@Sanchitbajaj02 changed according to your requirements. Please take a look.
Thank you

Reply:
Looks better now